PG.Commons/PG.Commons/Util/StringUtility.cs

```diff
@@ -96,5 +111,22 @@ private static List<string> SplitStringInternal(string s, char separator, bool s
             list.AddRange(from str in split where !IsNullEmptyOrWhiteSpace(str) select shouldClean ? str.Trim() : str);
             return list;
         }
+
+        /// <summary>
+        /// Removes any file extension form a file or filepath.
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentException"></exception>
+        public static string StripFileExtension(string fileName)
+        {
+            if (!StringUtility.HasText(fileName))
+            {
+                throw new ArgumentException("The provided argument is null or only consists of whitespace.",
+                    nameof(fileName));
+            }
+
+            return Path.ChangeExtension(fileName, null);
```

Comment:
Since we are using `IFileSystem` all over the place. Shouldn't have it here as well?

Reply:
As far as I'm aware `Path.ChangeExtension(string string)` does not physically interact with the file system, so I don't see the need to make it mockable (not that we could inject anything into the static constructor anyways, and not to mention that we'd basically be going against the idea of DI in the first place by injecting into a static class).